python/ray/dataframe/io.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from itertools import chain
+from io import BytesIO
+import os
+
+from pyarrow.parquet import ParquetFile
+import pandas as pd
+
+from .dataframe import from_pandas, ray, DataFrame
+from . import get_nprtitions
+
+
+def read_parquet(path, engine='auto', columns=None, **kwargs):
+    """Load a parquet object from the file path, returning a DataFrame.
+    Ray DataFrame only supports pyarrow engine for now.
+
+    Args:
+        path: The filepath of the parquet file.
+              We only support local files for now.
+        engine: Ray only support pyarrow reader. This argument doesn't do anything for now. 
+        kwargs: Pass into parquet's read_row_group function.
+    """
+    pf = ParquetFile(path)
+    n_row_groups = pf.metadata.num_row_groups
+    partition_per_row_group = get_nprtitions() // n_row_groups
+    ray_row_groups = [
+        _read_parquet_row_group.remote(path, columns, i,
+                                       partition_per_row_group, kwargs)
+        for i in range(n_row_groups)
+    ]
+    return _vertical_concat(ray.get(ray_row_groups))
+
+
+@ray.remote
+def _read_parquet_row_group(path, columns, row_group, npartitions, kwargs={}):
+    """Read a parquet row_group given file_path. This function returns
+    a Ray DataFrame
+    """
+    pf = ParquetFile(path)
+    df = pf.read_row_group(row_group, columns=columns, **kwargs).to_pandas()
+    return from_pandas(df, npartitions=npartitions)
```

Comment:
Why do we return `from_pandas` here? Won't that create a `ray.DataFrame` remotely? Wouldn't it be sufficient to just return the `pd.DataFrame` and use those in the constructor?

Reply:
This is by design because `pf.read_row_group` return a 5 million rows dataframe. This is to big. That's why we create ray df instead of pandas. I also don't think it's more efficient to serialize and deserialize a 5 million row df just to be used in the constructor again. We have `_vertial_concat` to join ray_df together now.